From 7ed759ad53c16047181b4a4f731d4d16f3fb3249 Mon Sep 17 00:00:00 2001
From: Todd Lipcon <todd@cloudera.com>
Date: Sat, 21 Apr 2012 17:05:41 -0700
Subject: [PATCH 084/795] MR1: CDH-5434. Fix ability to submit MR1 job to secure HA HDFS

Reason: Bug
Author: Todd Lipcon
(cherry picked from commit 3e9e1676568f2e68b81116325f9fa1551bbae571)
(cherry picked from commit cf7efa8aec1819ac8d940b457998c1e2b7ef6b40)
---
 .../hadoop/mapreduce/security/TokenCache.java      |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/src/mapred/org/apache/hadoop/mapreduce/security/TokenCache.java b/src/mapred/org/apache/hadoop/mapreduce/security/TokenCache.java
index c6a7a0b..c7d8db1 100644
--- a/src/mapred/org/apache/hadoop/mapreduce/security/TokenCache.java
+++ b/src/mapred/org/apache/hadoop/mapreduce/security/TokenCache.java
@@ -127,9 +127,7 @@ static void obtainTokensForNamenodesInternal(Credentials credentials,
         }
         Token<?> token = fs.getDelegationToken(delegTokenRenewer);
         if (token != null) {
-          Text fsNameText = new Text(fsName);
-          token.setService(fsNameText);
-          credentials.addToken(fsNameText, token);
+          credentials.addToken(token.getService(), token);
           LOG.info("Got dt for " + p + ";uri="+ fsName + 
                    ";t.service="+token.getService());
         }
-- 
1.7.0.4

